### ISSUE TITLE
ModelOptimizer

### ISSUE THREAD

**Issue by mpgirro** [mpgirro/stalla#47]
Placeholder for a future `ModelOptimizer` feature.

**Comment by rock3r** [comment]
What sorts of optimisation would you perform? Stuff like "if the googleplay value for X is identical to the itunes one, drop the googleplay one"?

**Comment by mpgirro** [comment]
If I remember correctly, the initial idea was about iTunes categories. E.g. the category `ItunesCategory.BOOKS` implies `ItunesCategory.ARTS` (because it's the parent category), so `ARTS` could be removed. Right now, I have two thoughts on this however:

1. I'd still keep `ARTS` now, because the order may be relevant. The Apple podcast directory for example only minds the first defined category, which might be `ARTS` on purpose, but the feed authors may still wish to add the subcategory for other directories. 
2. We do not check for duplicates in iTunes and Google Play categories iirc. This could be an optimisation.

**Comment by rock3r** [comment]
Agreed on the first point — for categories, the order matters and I wouldn't meddle with them. As for the example optimisation I had mentioned, I think that'd be the only sensible thing to do; something like "no point having a content:encoded if it's exactly the same as the rss description" would be good (but is a tad risky as you need to consider HTML and CDATA for asserting equivalence — for some platforms, HTML in the description is ignored if it's not in CDATA, in others it's ignored regardless)
